addons/point_of_sale/static/src/app/screens/product_screen/control_buttons/order_note_button/order_note_button.js

```diff
@@ -0,0 +1,55 @@
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { TextInputPopup } from "@point_of_sale/app/utils/input_popups/text_input_popup";
+import { usePos } from "@point_of_sale/app/store/pos_hook";
+import { makeAwaitable } from "@point_of_sale/app/store/make_awaitable_dialog";
+import { Component } from "@odoo/owl";
+
+export class GeneralNoteButton extends Component {
+    static template = "point_of_sale.GeneralNoteButton";
+    static props = {
+        label: { type: String, optional: true },
+        getter: { type: Function, optional: true },
+        setter: { type: Function, optional: true },
+        class: { type: String, optional: true },
+    };
+    static defaultProps = {
+        label: _t("General Note"),
+        getter: (order) => order.getGeneralNote(),
+        setter: (order, general_note) => order.setGeneralNote(general_note),
+        class: "",
+    };
+
+    setup() {
+        this.pos = usePos();
+        this.dialog = useService("dialog");
+    }
+    async onClick() {
+        const selectedOrder = this.pos.get_order();
+        const selectedNote = this.props.getter(selectedOrder);
+        const buttons = this.pos.models["pos.note"].getAll().map((note) => ({
+            label: note.name,
+            isSelected: selectedNote.split("\n").includes(note.name), // Check if the note is already selected
+        }));
+        const payload = await makeAwaitable(this.dialog, TextInputPopup, {
+            title: _t("Add %s", this.props.label),
+            buttons,
+            rows: 4,
+            startingValue: selectedNote,
+        });
+        if (
+            payload !== this.pos.getOrderChanges().lastGeneralNote &&
+            Object.keys(selectedOrder.last_order_preparation_change).length // Update existing order in kitchen
+        ) {
+            const categChange = [{ id: selectedOrder.id, count: 1, name: this.props.label }];
+            this.pos.isGeneralNote =
+                this.pos.isGeneralNote &&
+                !this.pos.isGeneralNote?.map((item) => item.id).includes(selectedOrder.id)
+                    ? this.pos.isGeneralNote.concat(categChange)
+                    : categChange;
```

Comment:
`isGeneralNote` sounds boolean to me, but why does it appear to be an array or undefined? Be consistent with the name. Also, what's the use of this global field?

Reply:
We have `getOrderChanges()` function which define the changes and based on that we show changes on the order button like 2 Food 1 Drink.... But all these changes are at orderline level if internal_note, qty,etc changes it will reflect change to that category.

But we need to send the changed order note to preparation display .
When we have no other changes we will have order button disabled but we want to send this order note changes as well which is of order level so we have found this way to allow sending even if only order note is changed.

Why here and not in `getOrderChanges` function?
We have payload here from where we can find the change  easily and the getOrderChanges function returns the changed lines so it was quite difficult to change the returned data format and to find the old order note there.

This is not a python field just on js side to detect and send when only order note changed